ckanext/dgu/theme/templates/package/resource_read.html

```diff
@@ -1,288 +1,163 @@
-      <a py:if="c.resource.get('url')" class="btn btn-danger resource-url-analytics resource-type-${c.resource.get('resource_type')}" href="${c.resource.get('url')}" onclick="${h.ga_download_tracking(c.resource, publisher.name)}">
-        <py:choose test="c.resource.get('resource_type')">
-          <py:when test="'listing'"><i class="icon-search"></i>&nbsp; View</py:when>
-          <py:when test="'service'"><i class="icon-search"></i>&nbsp; View</py:when>
-          <py:when test="'api'"><i class="icon-code"></i>&nbsp; API Endpoint</py:when>
-          <py:otherwise>
+    {% endif %}
+
+    {% with %}
+      {% set publisher = h.get_organization_from_resource(c.resource) %}
+    <div class="pull-right">
+      {% if c.resource.get('url') %}
+      <a class="btn btn-danger resource-url-analytics resource-type-{{c.resource.get('resource_type')}}" href="{{c.resource.get('url')}}" onclick="{{h.ga_download_tracking(c.resource, publisher.name)}}">
+        {% if c.resource.get('resource_type') == 'listing' %}
+          <i class="icon-search"></i>&nbsp; View
+        {% elif c.resource.get('resource_type') == 'service' %}
+          <py:when test="'service'"><i class="icon-search"></i>&nbsp; View
+        {% elif c.resource.get('resource_type') == 'api' %}
+          <i class="icon-code"></i>&nbsp; API Endpoint
+        {% else %}
             <div class="download">
-              <py:if test="not h.get_resource_wms(c.resource)">
+              {% if not h.get_resource_wms(c.resource) %}
                 <i class="icon-download"></i>&nbsp; Download
-              </py:if>
-              <py:if test="h.get_resource_wms(c.resource)">
+              {% else %}
                 <i class="icon-search"></i>&nbsp; Link
-              </py:if>
+              {% endif %}
             </div>
-          </py:otherwise>
-        </py:choose>
+        {% endif %}
       </a>
-      <py:with vars="cache_url, cache_timestamp = h.get_cache(c.resource)">
-       <a py:if="cache_url and not bool(h.get_resource_wms(c.resource))" class="btn btn-danger resource-url-analytics resource-type-${c.resource.get('resource_type')}" href="${cache_url}" onclick="${h.ga_download_tracking(c.resource, publisher.name, 'download-cache')}">
+      {% endif %}
+      {% endwith %}
+
+      {% set cache_url, cache_timestamp = h.get_cache(c.resource) %}
+      {% if cache_url and not h.get_resource_wms(c.resource) %}
+       <a class="btn btn-danger resource-url-analytics resource-type-${c.resource.get('resource_type')}" href="${cache_url}" onclick="${h.ga_download_tracking(c.resource, publisher.name, 'download-cache')}">
         <div class="download js-tooltip" data-placement="bottom" data-original-title="Cached by data.gov.uk on: ${cache_timestamp.strftime('%d/%m/%Y') if cache_timestamp else 'unknown'}">
-          <py:if test="not bool(h.get_resource_wms(c.resource))">
+          {% if not h.as_bool(h.get_resource_wms(c.resource)) %}
```

Comment:
I think the bool() casting was unnecessary as it would be done by the not anyway, so let's drop the h.as_bool() which implies something more complicated.